### PR TITLE
flytekit 1.3.0

### PR DESCRIPTION
**Pull request by bstadlbauer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bstadlbauer** [comment]
@conda-forge-admin, please rerender

**Comment by bstadlbauer** [comment]
@cosmicBboy I've investigated the dependency issue you've been seeing in https://github.com/conda-forge/flytekit-feedstock/pull/31

While the original error message was related to `python_abi`, I've seen this being a symptom of other conflicts that occur between packages. 

A diff of the [dependencies in setup.py](https://github.com/flyteorg/flytekit/blob/v1.3.0/setup.py) showed that the `meta.yaml` still had a `protobuf >=3.6.1,<4` dependency. After removing that, I could locally create an environment.  

In case you ever need it - the workflow I've used to check whether resolving works is 
1. Check out the branch from the bot
2. Install [`conda-lock`](https://github.com/conda-incubator/conda-lock)
3. Try to create a lockfile from the `meta.yaml` (this resolves all dependencies) using `conda-lock lock -f recipe/meta.yaml -p linux-64`. 
This showed me the same error as on CI and was resolved after removing `protobuf`

@asford In case you have a better way (or a way to directly surface conflicts), would appreciate any help :-)

**Comment by bstadlbauer** [comment]
@cosmicBboy I had to pin `python >=3.8,<3.11` to avoid running into https://github.com/flyteorg/flyte/issues/3181

**Comment by bstadlbauer** [comment]
@asford I think things would be read from my end

**Comment by cosmicBboy** [comment]
@asford should we merge this now, or do you want to hold off until we discuss longer term maintenance strategy?

File: recipe/meta.yaml
Comment (asford) [inline recipe/meta.yaml:46]:
```suggestion
    - grpcio >=1.50.0,<2.0
    - grpcio-status >=1.50.0,<2.0
```

**Review by asford — APPROVED** [review]
LGTM

**Comment by asford** [comment]
> @cosmicBboy I've investigated the dependency issue you've been seeing in #31
> 
> While the original error message was related to `python_abi`, I've seen this being a symptom of other conflicts that occur between packages.
> 
> A diff of the [dependencies in setup.py](https://github.com/flyteorg/flytekit/blob/v1.3.0/setup.py) showed that the `meta.yaml` still had a `protobuf >=3.6.1,<4` dependency. After removing that, I could locally create an environment.
> 
> In case you ever need it - the workflow I've used to check whether resolving works is
> 
> 1. Check out the branch from the bot
> 2. Install [`conda-lock`](https://github.com/conda-incubator/conda-lock)
> 3. Try to create a lockfile from the `meta.yaml` (this resolves all dependencies) using `conda-lock lock -f recipe/meta.yaml -p linux-64`.
>    This showed me the same error as on CI and was resolved after removing `protobuf`
> 
> @asford In case you have a better way (or a way to directly surface conflicts), would appreciate any help :-)

Yes, this is definitely a bit of a hole in the conda-forge maintainer tooling.
We have an internal tool that performs some fancy footwork to help with this kind of "interlocked dependency" resolution, 
but that shouldn't be needed here.
we generally just checkout and install `conda` and `mamba`, so the workflow is:

1. Checkout recipe the branch via the needful.
2. Install `conda`, `mamba`, and `grayskull`.
3. Run `conda mambabuild recipe -c conda-forge` to test build resolution, which uses the mamba solver and is quite a bit faster.
4. After getting a successful local build, run `build-locally.py` to verify the build or push to CI for verification.

Often, it's nice to `grayskull pypi <package>` to re-generate the recipe from the upstream package, then inspect the differences between that and the existing recipe. This is essentially what the autotick bot is doing, but I don't know if this is exposed as a cli component yet.

**Comment by asford** [comment]
Ah, just looked at #31. 
This particular mismatch was actually found by the grayskull dep analysis...

Packages found in the meta.yaml but not found by grayskull:
...
protobuf >=3.6.1,<4

**Comment by bstadlbauer** [comment]
@asford Thank you very much! That helped a lot 👍

**Pull request merged**
